pkg/appendable/csv_handler.go

```diff
@@ -17,12 +17,18 @@ type CSVHandler struct {
 	io.ReadSeeker
 }
 
-func (c CSVHandler) Synchronize(f *IndexFile) error {
-
+func (c CSVHandler) Synchronize(f *IndexFile, headerData []string) error {
```

Comment:
The headers should be able to be determined from the index file itself since they're already parsed.

https://github.com/kevmo314/appendable/blob/165a9ef6934b8b1df941f3d3637ccaa3bdb57be3/pkg/protocol/protocol.go#L49
https://github.com/kevmo314/appendable/blob/165a9ef6934b8b1df941f3d3637ccaa3bdb57be3/pkg/appendable/index_file.go#L16

Basically, when parsing the lines, if the `Indexes` are not present then parse the line as `Indexes` (in other words, read the headers). If the `Indexes` are present, then the line represents a data line. This will allow the handler to not need to take in the headers as an argument.

This will be true in the B+ tree too (although it's called the MetaPage) there.

Reply:
Damn, this is super big brain. Makes things so much nicer. Thanks Kevin!